src/index.ts

```diff
@@ -13,13 +12,26 @@ function VitePluginComponents(options: Options = {}): Plugin {
     name: 'vite-plugin-components',
     enforce: 'post',
     configResolved(config) {
-      if (config.plugins.find(i => i.name === 'vite-plugin-vue2'))
+      if (config.plugins.find(i => i.name === 'vite-plugin-vue2')) {
         options.transformer = options.transformer || 'vue2'
+      }
+      else if (config.plugins.find(i => i.name === 'vite-plugin-svelte')) {
+        options.transformer = options.transformer || 'svelte3'
+
+        if (options.extensions) {
+          if (typeof options.extensions === 'string') {
+            // convert to array
+            options.extensions = [options.extensions]
+          }
+          options.extensions.push('svelte')
+        }
+        else {
+          options.extensions = 'svelte'
+        }
```

Comment:
Move this to `resolveOptions`, use `options.transformer` to check is svelte or not.